### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by anasm11**
moved usereducer for products to context (ProductsContext.js) and other minor changes (homepage functionality,added custom data)

### CODE REVIEW

File: src/context/CartContext.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      },
+      data: {
+        action: {
+          type: "increment"
+        }
+      }
+
+    })
+    setCartItems(res.data.cart)
+    console.log(res.data, 'p')
```
Comment (Adarsh-Patel-Dev) [inline src/context/CartContext.js]: remove console.log

File: src/context/WishListContext.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import axios from "axios"
+import { useContext, useState, createContext } from "react"
+
+const WishListContext = createContext()
+
+
```
Comment (Adarsh-Patel-Dev) [inline src/context/WishListContext.js]: remove extra spaces here

**Review by Adarsh-Patel-Dev — APPROVED** [review]
suggested minor changes
Rest code is good to merge

File: src/components/WishList.jsx [hunk 3]
```diff
@@ -1,39 +1,23 @@
 import React from 'react'
 import '../css/WishList.css'
-import { productsData, IcRoundFavorite, IcRoundShare, IcSharpMoreVert } from '../assets/index'
+import Card from './Card'
+import { useCart } from '../context/CartContext'
+import { useWishListContext } from '../context/WishListContext'
```
Comment (yaswanthmyneni) [inline src/components/WishList.jsx]: suggestion: export all contexts from index.js in context folder

File: src/context/CartContext.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    const res = await axios({
+      method: "post",
+      url: "/api/user/cart",
+      headers: { authorization: localStorage.getItem('token') },
+      data: {
+        product: product
+      },
+    });
+
+    console.log(res.data.cart, 'added')
```
Comment (yaswanthmyneni) [inline src/context/CartContext.js]: changes: remove the console.log when pushing code to Github

File: src/context/ProductsContext.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        console.log(productsState.filterPrice,'fromfilter')
+        var prodcts = allProducts.filter((prdct) => {
+            for (let i in productsState.categoryFilter) {
+                if (prdct.categoryName === productsState.categoryFilter[i])
+                    return true
+            }
+            return false
+        })
+    
+         let x=prodcts.filter((item)=>parseInt(item.rating)>=parseInt(productsState.filterRating))
```
Comment (yaswanthmyneni) [inline src/context/ProductsContext.js]: changes: use proper variable names everywhere

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes

**Pull request merged**